consul/base.py

```diff
@@ -138,21 +138,33 @@ def callback(
         is_json=False,
         index=False,
         one=False,
-        allow_404=True):
+        allow_404=True,
+        decode=None,
+        is_id=False):
```

Comment:
still brainstorming... perhaps something similar to how the Check. methods are broken out so that their apis are unambiguous. something like:

```
# for is_json
CB.json(map=None, allow_404=True, with_index=False, one=False, decode_key=None, is_id=False)

# for is_200
CB.bool(map=None, allow_404=True, with_index=False)

# let's data fall through unmolested (is this case even used)?
CB.plain(map=None, allow_404=True, with_index=False)
```

Noting: `one` is only makes sense with the `is_json` option. Renamed `index` to `with_index` and `decode` to `decode_key` for clarity.